src/Microsoft.PowerShell.Commands.Management/commands/management/TestConnectionCommand.cs

```diff
@@ -275,53 +294,105 @@ protected override void ProcessRecord()
         protected override void StopProcessing()
         {
             _sender?.SendAsyncCancel();
+            cancellationTokenSource.Cancel();
         }
 
         #region ConnectionTest
 
+        private void SetCountForTcpTest()
+        {
+            if (Repeat.IsPresent)
+            {
+                Count = int.MaxValue;
+            }
+            else if (!MyInvocation.BoundParameters.ContainsKey("Count"))
+            {
+                Count =  1;
+            }
+        }
+
         private void ProcessConnectionByTCPPort(string targetNameOrAddress)
         {
+            SetCountForTcpTest();
+
             if (!TryResolveNameOrAddress(targetNameOrAddress, out _, out IPAddress? targetAddress))
             {
                 return;
             }
 
-            TcpClient client = new TcpClient();
-
-            try
+            for (var i = 1; i <= Count; i++)
             {
-                Task connectionTask = client.ConnectAsync(targetAddress, TcpPort);
-                string targetString = targetAddress.ToString();
+                long latency = 0;
+                SocketError status = SocketError.SocketError;
+
+                Stopwatch stopwatch = new Stopwatch();
+
+                using TcpClient client = new TcpClient();
 
-                for (var i = 1; i <= TimeoutSeconds; i++)
+                try
                 {
-                    Task timeoutTask = Task.Delay(millisecondsDelay: 1000);
-                    Task.WhenAny(connectionTask, timeoutTask).Result.Wait();
+                    stopwatch.Start();
 
-                    if (timeoutTask.Status == TaskStatus.Faulted || timeoutTask.Status == TaskStatus.Canceled)
+                    if (client.ConnectAsync(targetAddress, TcpPort).Wait(TimeoutSeconds * 1000, cancellationToken))
                     {
-                        // Waiting is interrupted by Ctrl-C.
-                        WriteObject(false);
-                        return;
+                        latency = stopwatch.ElapsedMilliseconds;
+                        status = SocketError.Success;
+                    }
+                    else
+                    {
+                        status = SocketError.TimedOut;
                     }
+                }
+                catch (AggregateException ae)
+                {
+                    ae.Handle((ex) =>
+                    {
+                        if (ex is SocketException)
```

Comment:
We should process TaskCanceledException. See #11517.

Reply:
I see what you mean! I've tried to line up the logic with the changes in #11517. Is this what you're thinking? 

```csharp
catch (AggregateException ae)
{
    ae.Handle((ex) =>
    {
        if (ex is TaskCanceledException)
        {
            throw new PipelineStoppedException();
        }
        if (ex is SocketException)
        {
            SocketException? socketException = ex as SocketException;
            status = socketException!.SocketErrorCode;
            return true;
        }
```
Let me know your thoughts! 😁

Reply:
Looks good.

Reply:
Great! I've pushed that change 😊